src/HttpCache/VarnishPurger.php

```diff
@@ -48,10 +48,10 @@ public function purge(array $iris)
             return sprintf('(^|\,)%s($|\,)', preg_quote($iri));
         }, $iris);
 
-        $regex = isset($parts[1]) ? sprintf('(%s)', implode(')|(', $parts)) : $parts[0];
+        $regex = count($parts) > 1 ? sprintf('(%s)', implode(')|(', $parts)) : array_shift($parts);
```

Comment:
`current` might work, not sure about the perf reward though. `array_shift` does put cursor in position 0 every call.